server.js

```diff
@@ -1,30 +1,129 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
-
-// Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
-
-// Start defining your routes here
+import express from 'express';
+import cors from 'cors';
+import mongoose from 'mongoose';
+
+const mongoUrl = process.env.MONGO_URL || 'mongodb://localhost/happyThoughts';
+mongoose.connect(mongoUrl, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+  useCreateIndex: true,
+});
+mongoose.Promise = Promise;
+
+const port = process.env.PORT || 8080;
+const app = express();
+
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    reguired: [true, 'Message required'],
+    minlength: 5,
+    maxlength: 140,
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+  },
+});
+
+const Thought = mongoose.model('Thought', thoughtSchema);
+
+app.use(cors());
+app.use(express.json());
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.send('Hello world');
+});
+
+app.get('/thoughts', async (req, res) => {
+  try {
+    const thoughts = await Thought.find();
+    const sortThoughts = thoughts.sort((a, b) => b.createdAt - a.createdAt);
```

Comment:
Your code should be:
```jsx
const thoughts = await Thought.find().sort({ createdAt: -1 }).limit(20)
``` 
It is important to do it this way because if you do your sorting and limiting manually on the server you already downloading a loooooooooooooooooooot more data from the database than you need. what on bigger projects can be heaven and hell situation. You are using your server resources to sort and return only microscopic parts of it to your user.